lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -601,6 +797,33 @@ describe('ContentMetadataComponent', () => {
             const basicPropertiesGroup = fixture.debugElement.query(By.css('.adf-metadata-grouped-properties-container mat-expansion-panel'));
             expect(basicPropertiesGroup).toBeDefined();
         });
+
+        it('should revert changes for general info panel on cancel', () => {
+            const spy = spyOn(component as any, 'getProperties').and.returnValue(of(expectedNode));
+            component.onCancelGeneralInfoEdit();
+            expect(spy).toHaveBeenCalled();
+        });
+
+        it('should revert changes for getGroupedProperties panel on cancel', () => {
+            spyOn(contentMetadataService, 'getGroupedProperties');
+            component.ngOnChanges({ node: new SimpleChange(node, expectedNode, false) });
+            component.onCancelGroupEdit({} as any);
+            expect(contentMetadataService.getGroupedProperties).toHaveBeenCalledWith(expectedNode, 'custom-preset');
+        });
+
+        it('should revert changes for categories panel on cancel', () => {
+            const spy = spyOn(component as any, 'loadCategoriesForNode');
```

Comment:
you should not use 'as any' just to be able to call private functions (like getProperties). You should call private things by public functions etc.

Reply:
done